pkg/handler/v2/external_cluster/eks.go

```diff
@@ -421,7 +422,26 @@ func patchEKSCluster(oldCluster, newCluster *apiv2.ExternalCluster, secretKeySel
 	}
 	_, err = client.EKS.UpdateClusterVersion(&updateInput)
 	if err != nil {
-		return nil, err
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case eks.ErrCodeInvalidParameterException:
+				return nil, errors.New(aerr.Message())
+			case eks.ErrCodeClientException:
+				return nil, errors.New(aerr.Message())
+			case eks.ErrCodeServerException:
+				return nil, errors.New(aerr.Message())
+			case eks.ErrCodeResourceInUseException:
+				return nil, errors.New(aerr.Message())
+			case eks.ErrCodeResourceNotFoundException:
+				return nil, errors.New(aerr.Message())
+			case eks.ErrCodeInvalidRequestException:
+				return nil, errors.New(aerr.Message())
+			default:
+				return nil, err
```

Comment:
What were the criteria for choosing these exact errors to unwrap? Why not do this for _all_ AWS errors?